### DISCUSSION TITLE
Which models can be used text embedding generation?

### DISCUSSION THREAD

**Discussion by ardacetinkaya** [community/community#135548] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I am checking those available models with .NET Semantic Kernal APIs. 

When I am trying text embedding generation for OpenAI GPT-4o, I am getting the following error. 
Is it normal?

> "error": {
        "message": "Incorrect API key provided: github_p*********************************************************************************mDd2. You can find your API key at https://platform.openai.com/account/api-keys.",
        "type": "invalid_request_error",
        "param": null,
        "code": "invalid_api_key"
    }

**Comment by matthewisabel** [comment]
Which endpoint did you use to configure the completions? Was it "https://models.inference.ai.azure.com"?

**Comment by nick863** [comment, marked as answer]
gpt-4o does not support text embeddings. For text embeddings, please try `text-embedding-3-large` and `text-embedding-3-small`.
